rosidl_typesupport_c/QUALITY_DECLARATION.md

```diff
@@ -0,0 +1,123 @@
+### Contributor Origin [2.ii]
+ This package has a confirmation of contributor origin policy, which can be found in [CONTRIBUTING](../CONTRIBUTING.md)
+
+### Peer Review Policy [2.iii]
+
+ Following the recommended guidelines for ROS Core packages, all pull request have at least 1 peer review.
+
+### Continuous Integration [2.iv]
+
+All pull request must pass CI on all [tier 1 platforms](https://www.ros.org/reps/rep-2000.html#support-tiers)
+
+Currently nightly results can be seen here:
+* [linux-aarch64_release](https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_release/lastBuild/testReport/rosidl_typesupport_c/)
+* [linux-arm64_release](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/rosidl_typesupport_c/)
+* [mac_osx_release](https://ci.ros2.org/view/nightly/job/nightly_osx_release/lastBuild/testReport/rosidl_typesupport_c/)
+* [windows_release](https://ci.ros2.org/view/nightly/job/nightly_win_rel/lastBuild/testReport/rosidl_typesupport_c/)
+
+## Documentation [3]
+
+### Feature Documentation [3.i]
+
+`rosidl_typesupport_c` does not have any feature documentation and it will need to be added for higher quality levels.
+
+### Public API Documentation [3.ii]
+
+`rosidl_typesupport_c` does not currently have public API documentation.
+
+### License [3.iii]
+
+The license for `rosidl_typesupport_c` is Apache 2.0, and a summary is in each source file, the type is declared in the `package.xml` manifest file, and a full copy of the license is in the [LICENSE](./LICENSE) file.
+
+There is an automated test which runs a linter that ensures each file has a license statement.
+
+### Copyright Statements [3.iv]
+
+The copyright holders each provide a statement of copyright in each source code file in `rosidl_typesupport_c`.
+
+There is an automated test which runs a linter that ensures each file has at least one copyright statement.
+
+Most recent test results can be found [here](http://build.ros2.org/view/Epr/job/Epr__rosidl_typesupport__ubuntu_bionic_amd64/lastBuild/testReport/rosidl_typesupport_c/)
+
+
+## Testing [4]
+
+### Feature Testing [4.i]
+
+There are currently no public features undergoing tests.
+
+### Public API Testing [4.ii]
+
+There are currently no tests for the public API.
+
+### Coverage [4.iv]
+
+`rosidl_typesupport_c` does not currently track test coverage.
+
+### Performance [4.iv]
+
+`rosidl_typesupport_c` does not currently have performance tests.
+
+### Linters and Static Analysis [4.v]
+
+`rosidl_typesupport_c` uses and passes all the standard linters and static analysis tools for a C++ package as described in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#linters).
+
+## Dependencies [5]
+
+### Direct Runtime ROS Dependencies [5.i]
+`rosidl_typesupport_c` has the following ROS dependencies:
+* `rcpputils`
+* `rosidl_runtime_c`
+* `rosidl_typesupport_connext_c`
+* `rosidl_typesupport_interface`
+* `rosidl_typesupport_introspection_c`
+
```

Comment:
How about adding an explicit note that "`rosidl_typesupport_c` does not have any direct runtime "non-ROS" dependencies."

Reply:
Added section 5.iii for that.